telemetry/telemetry_test.go

```diff
@@ -361,6 +361,14 @@ func TestCopyRecords(t *testing.T) {
 		_ = os.RemoveAll(dir)
 	}()
 
+	_, err = os.Stat(telemetrySpoolDir)
```

Comment:
If you are creating the directory, then you need to clean it up as well.
NOTE: The `TestLogRecord` test below also expects the directory to exist so will need a similar or shared fix.

The other approach I considered was skipping the test if the spool directory is missing:

```
diff --git a/telemetry/telemetry_test.go b/telemetry/telemetry_test.go
index f4da46aacbe3..65714267d7d9 100644
--- a/telemetry/telemetry_test.go
+++ b/telemetry/telemetry_test.go
@@ -347,6 +347,10 @@ func TestCopyRecords(t *testing.T) {
                t.Skip("Not running as 'root', skipping record copy test")
        }

+       if exists, err := utils.FileExists(telemetrySpoolDir); err == nil && !exists {
+               t.Skip("Not running due to missing telemetry bundle, skipping record copy test")
+       }
+
        url = "http://www.google.com"
        if err := telem.SetTelemetryServer(url, tid, policy); err != nil {
                t.Fatalf("Setting telemetry server (%q) should not return error: %s\n", url, err)
@@ -387,6 +391,10 @@ func TestLogRecord(t *testing.T) {
        if !utils.IsRoot() {
                t.Skip("Not running as 'root', skipping log record test")
        }
+       if exists, err := utils.FileExists(telemetrySpoolDir); err == nil && !exists {
+               t.Skip("Not running due to missing telemetry bundle, skipping log record test")
+       }
+
        if err := telem.LogRecord("success", 2, "Hello"); err != nil {
                t.Logf("TestLogRecord failed with %s", err)
        }
```

But you get more coverage with your solution.

Reply:
Okay , I think I will clean up the directory with a defer as the code gets tested